### PR TITLE
feat/ui-stake-helpers

### PR DESCRIPTION
**Pull request by foodaka**

### CODE REVIEW

File: contracts/misc/StakedTokenDataProvider.sol [hunk 1]
```diff
@@ -47,6 +53,9 @@ contract StakedTokenDataProvider is IStakedTokenDataProvider {
    * @param ethUsdPriceFeed The address of ETH price feed (USD denominated, with 8 decimals)
    * @param aavePriceFeed The address of AAVE price feed (ETH denominated, with 18 decimals)
    * @param bptPriceFeed The address of StakedBpt price feed (ETH denominated, with 18 decimals)
+   * @param bptWstETH The address of stETH token
+   * @param bptWstETHPriceFeed The address of stEth price feed (ETH denominated, with 18 decimals)
+
```
Comment (miguelmtzinf) [inline contracts/misc/StakedTokenDataProvider.sol]: remove this breakline

File: contracts/misc/StakedTokenDataProvider.sol [hunk 2]
```diff
@@ -47,6 +53,9 @@ contract StakedTokenDataProvider is IStakedTokenDataProvider {
    * @param ethUsdPriceFeed The address of ETH price feed (USD denominated, with 8 decimals)
    * @param aavePriceFeed The address of AAVE price feed (ETH denominated, with 18 decimals)
    * @param bptPriceFeed The address of StakedBpt price feed (ETH denominated, with 18 decimals)
+   * @param bptWstETH The address of stETH token
+   * @param bptWstETHPriceFeed The address of stEth price feed (ETH denominated, with 18 decimals)
```
Comment (miguelmtzinf) [inline contracts/misc/StakedTokenDataProvider.sol]:
```suggestion
   * @param bptWstETH The address of wstETH token
   * @param bptWstETHPriceFeed The address of wstEth price feed (ETH denominated, with 18 decimals)
```

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 3]
```diff
@@ -17,6 +17,7 @@ interface IStakedTokenDataProvider {
     uint256 stakeApy;
     uint128 distributionPerSecond;
     uint256 distributionEnd;
+    bool inPostSlashingPeriod;
```
Comment (miguelmtzinf) [inline contracts/interfaces/IStakedTokenDataProvider.sol]:
```suggestion
    uint128 distributionPerSecond;
    bool inPostSlashingPeriod;
    uint256 distributionEnd;
```
moving up this variable would be more efficient -> `distributionPerSecond` and `inPostSlashingPeriod` in the same slot.

File: contracts/interfaces/IStakedTokenDataProvider.sol
Comment (miguelmtzinf) [inline contracts/interfaces/IStakedTokenDataProvider.sol:1]: Clarify the format and unit denomination of oracles on every @param please.

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 4]
```diff
@@ -59,83 +54,67 @@ interface IStakedTokenDataProvider {
...
-   * @notice Returns user data of Staked Aave
-   * @param user The address of the user
-   * @return stkAaveData An object with StkAave data
-   * @return stkAaveUserData An object with user data of StkAave
+   * @notice Retrieves data for a batch of staked assets given their addresses and the addresses of their associated oracles.
+   * @param stakedTokens An array of addresses representing the staked tokens.
+   * @param oracleAddresses An array of addresses representing the oracles associated with the staked tokens.
+   * @return stakedData An array containing data about the staked assets.
+   * @return prices An array containing the latest price data from the oracles for the staked tokens.
+   * @return ethPrice An array containing the latest ETH price for the staked tokens.
```
Comment (miguelmtzinf) [inline contracts/interfaces/IStakedTokenDataProvider.sol]: does not need to be an array, it's just the ETH/USD price, same for all staked tokens

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 5]
```diff
@@ -59,83 +54,67 @@ interface IStakedTokenDataProvider {
...
   /**
-   * @notice Returns user data of Staked Aave
-   * @param user The address of the user
-   * @return stkAaveData An object with StkAave data
-   * @return stkAaveUserData An object with user data of StkAave
+   * @notice Retrieves data for a batch of staked assets given their addresses and the addresses of their associated oracles.
+   * @param stakedTokens An array of addresses representing the staked tokens.
+   * @param oracleAddresses An array of addresses representing the oracles associated with the staked tokens.
+   * @return stakedData An array containing data about the staked assets.
+   * @return prices An array containing the latest price data from the oracles for the staked tokens.
```
Comment (miguelmtzinf) [inline contracts/interfaces/IStakedTokenDataProvider.sol:89]: why is this array needed? `stakedData` contains the price of the reward already iirc

File: contracts/interfaces/IStakedTokenDataProvider.sol [hunk 6]
```diff
@@ -59,83 +54,67 @@ interface IStakedTokenDataProvider {
...
+   * @param oracleAddresses An array of addresses representing the oracles associated with the staked tokens.
+   * @param userAddresses An array of addresses representing the users whose data should be retrieved.
+   * @return stkAaveData An array containing data about the staked assets.
+   * @return stakedUserData An array containing user-specific data about the staked assets.
    */
-  function getStkBptAaveUserData(address user)
+  function getStakedUserDataBatch(
+    address[] calldata stakedTokens,
+    address[] calldata oracleAddresses,
+    address[] calldata userAddresses
```
Comment (miguelmtzinf) [inline contracts/interfaces/IStakedTokenDataProvider.sol]: Do you think an array of users is necessary? UI will probably query for 1 single user at a time.

File: contracts/misc/StakedTokenDataProvider.sol [hunk 7]
```diff
@@ -19,127 +19,120 @@ contract StakedTokenDataProvider is IStakedTokenDataProvider {
...
-  address public immutable override STAKED_BPT;
-
   uint256 private constant SECONDS_PER_YEAR = 365 days;
 
   uint256 private constant APY_PRECISION = 10000;
 
   /**
    * @dev Constructor
-   * @param aave The address of the AAVE token
+   * @param aave The address of the StkAAVE token
```
Comment (miguelmtzinf) [inline contracts/misc/StakedTokenDataProvider.sol]:
```suggestion
   * @param aave The address of the AAVE token
```

**Comment by grothem** [comment]
Link T-13575

**Comment by foodaka** [comment]
This will also impact the aave utilities, and the frontend integration

**Pull request closed without merging**